### PR TITLE
Enable masked systemd services in cloudapi (HMS-3661)

### PR DESCRIPTION
**Pull request by kingsleyzissou**
This pull request includes:
This PR adds support for `masked` systemd services via the cloud API.

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

Depends on:
https://github.com/osbuild/images/pull/464

### CODE REVIEW

File: internal/cloudapi/v2/compose_test.go [hunk 1]
```diff
@@ -78,6 +78,7 @@ func GetTestBlueprint() blueprint.Blueprint {
 		Services: &blueprint.ServicesCustomization{
 			Enabled:  []string{"sshd"},
 			Disabled: []string{"cleanup"},
+			Masked:   []string{"firewalld"},
```
Comment (bcl) [inline internal/cloudapi/v2/compose_test.go:81]: The `testBlueprint` above also needs to have `"masked"` ... added to it.

File: internal/weldr/api_test.go [hunk 2]
```diff
@@ -629,6 +629,7 @@ func TestBlueprintsCustomizationInfoToml(t *testing.T) {
 			Services: &blueprint.ServicesCustomization{
 				Enabled:  []string{"sshd", "cockpit.socket", "httpd"},
 				Disabled: []string{"postfix", "telnetd"},
+				Masked:   []string{"firewalld"},
```
Comment (bcl) [inline internal/weldr/api_test.go:632]: The `testBlueprint` above needs to have `"masked"` ... added to it.

**Comment by thozza** [comment]
Converted to draft, since it needs the change to land in `images` first.

File: internal/blueprint/blueprint.go [hunk 3]
```diff
@@ -48,9 +48,8 @@ type Container struct {
 	Source string `json:"source,omitempty" toml:"source"`
 	Name   string `json:"name,omitempty" toml:"name,omitempty"`
 
-	TLSVerify           *bool   `json:"tls-verify,omitempty" toml:"tls-verify,omitempty"`
```
Comment (bcl) [inline internal/blueprint/blueprint.go:51]: I think the container related changes and the mask changes should be split into separate commits. I also have other concerns about the blueprint field name changes, but since it's already in images that boat may have sailed ;)

**Review by thozza — COMMENTED** [review]
Changes look good to me, but I would propose to merge the first two commits into a single one (update of images to v0.41.0 and changes to the blueprint). The reason is that otherwise, the commit updating the images version will result in composer not being able to compile. This could impact potential bisecting in the future.

File: internal/blueprint/blueprint.go (hunk 3 above)
Comment (kingsleyzissou) [reply internal/blueprint/blueprint.go:51]: I had that initially. But then it meant that one of the commits would still be broken and wouldn't compile 🫠 so I squashed them together. Happy to change it back though

**Review by thozza — APPROVED** [review]
